.github/workflows/test.yml

```diff
@@ -0,0 +1,69 @@
+name: Smoke Tests
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  klippy_testing:
+    name: Klippy Tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        klipper_repo:
+          - klipper3d/klipper
+          - DangerKlippers/danger-klipper
+    steps: 
+      - name: Checkout shaketune
+        uses: actions/checkout@v4
+        with:
+          path: shaketune
+      - name: Checkout Klipper
+        uses: actions/checkout@v4
+        with:
+          path: klipper
+          repository: ${{ matrix.klipper_repo }}
+          ref: master
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install build-essential
+      - name: Build klipper dict
+        run: |
+          pushd klipper
+          cp ../shaketune/ci/smoke-test/klipper-smoketest.kconfig .config
+          make olddefconfig
+          make out/compile_time_request.o
+          popd
+      - name: Setup klippy env
+        run: |
+          python3 -m venv --prompt klippy klippy-env
+          ./klippy-env/bin/python -m pip install -r klipper/scripts/klippy-requirements.txt
```

Comment:
This was a good suggestion, but the patch provided does not remove the second invocation to pip.

Reply:
Unfortunately, this had to be reverted, as there is a dependency conflict between the two.
I've chosen to leave the two invocations in, as this will be like the vast majority of end-user configurations.

In the future, It would probably be best to use a nonspecific dependency for packages that are shared with (or have specific deps for packages that are) shared with klipper.
see: https://github.com/Laikulo/shaketune/actions/runs/9688597940/job/26735455067